constructor/preconda.py

```diff
@@ -244,6 +252,13 @@ def write_conda_meta(info, dst_dir, final_urls_md5s, user_requested_specs=None):
         fh.write("\n".join(builder))
 
 
+def write_frozen(freeze_info, dst_dir):
+    if freeze_info and "conda" in freeze_info:
```

Comment:
This would result in you checking for `freeze_info`, once before the call and once during the call.